packit_service/models.py

```diff
@@ -2329,6 +2330,49 @@ def add_scan(self, task_id: int) -> "OSHScanModel":
             session.add(scan)
             return scan
 
+    @contextmanager
+    def add_scan_transaction(self) -> Generator["OSHScanModel"]:
+        """
+        Context manager that creates a ScanModel upon entering the context,
+        provides a corresponding instance of `ScanModel` to be updated within the context
+        and commits the changes upon exiting the context, all within a single transaction.
+
+        This locking mechanism is working on the assumption that just a single scan model
+        for build can exist.
+
+        raise: IntegrityError if the scan model already exists
+        """
+        session = singleton_session or Session()
+        session.begin()
+
+        try:
+            if not self.scan:
+                scan = OSHScanModel()
+                scan.copr_build_target = self
+                session.add(scan)
+                session.commit()
+            else:
+                raise IntegrityError("Scan model already exists")
```

Comment:
```suggestion
            scan = OSHScanModel()
            scan.copr_build_target = self
            session.add(scan)
            session.commit()
```

The condition is redundant, if the scan already exists, `IntegrityError` will be raised either way.

Reply:
🤔 if there is already the scan, I don't `commit` anything and no exception will be raised before me yielding a scan. Since I don't want to submit a new scan or check for a `None` scan in the caller, I need to raise the exception here.